inherited variadic ctor

```
  In file included from amici/include/amici/misc.h:5:0,
                   from amici/amici_wrap.cxx:5746:
  amici/include/amici/exception.h:134:25: warning: the ellipsis in ‘amici::AmiException::AmiException(const char*, ...)’ is not inherited [-Winherited-variadic-ctor]
       using AmiException::AmiException;
                           ^~~~~~~~~~~~
  amici/include/amici/exception.h:23:14: note: ‘amici::AmiException::AmiException(const char*, ...)’ declared here
       explicit AmiException(char const* fmt, ...);
                ^~~~~~~~~~~~
```

--- a/include/amici/exception.h
+++ b/include/amici/exception.h
@@ -20,6 +20,13 @@ public:
      * @param fmt error message with printf format
      * @param ... printf formatting variables
      */
+    AmiException();
+
+    /**
+     * @brief Constructor with printf style interface
+     * @param fmt error message with printf format
+     * @param ... printf formatting variables
+     */
     explicit AmiException(char const* fmt, ...);
 
     /**
@@ -39,6 +46,14 @@ public:
      * @param nMaxFrames number of frames to go back in stacktrace
      */
     void storeBacktrace(int nMaxFrames);
+
+  protected:
+    /**
+     * @brief Store the provided message
+     * @param fmt error message with printf format
+     * @param argptr pointer to variadic argument list
+     */
+    void storeMessage(const char *fmt, va_list argptr);
 
 private:
     std::array<char, 500> msg_;
@@ -131,7 +146,13 @@ class IntegrationFailureB : public AmiException  {
  */
 class SetupFailure : public AmiException {
   public:
-    using AmiException::AmiException;
+    /**
+     * @brief Constructor with printf style interface
+     * @param fmt error message with printf format
+     * @param ... printf formatting variables
+     */
+    explicit SetupFailure(char const* fmt, ...);
+
 };
 
 

--- a/src/exception.cpp
+++ b/src/exception.cpp
@@ -8,12 +8,18 @@
 
 namespace amici {
 
-AmiException::AmiException(const char *fmt, ...) {
+AmiException::AmiException()
+{
+    storeBacktrace(12);
+}
+
+AmiException::AmiException(const char *fmt, ...)
+    : AmiException()
+{
     va_list ap;
     va_start(ap, fmt);
-    vsnprintf(msg_.data(), msg_.size(), fmt, ap);
+    storeMessage(fmt, ap);
     va_end(ap);
-    storeBacktrace(12);
 }
 
 const char *AmiException::what() const noexcept {
@@ -27,6 +33,11 @@ const char *AmiException::getBacktrace() const {
 void AmiException::storeBacktrace(const int nMaxFrames) {
     snprintf(trace_.data(), trace_.size(), "%s",
              backtraceString(nMaxFrames).c_str());
+}
+
+void AmiException::storeMessage(const char *fmt, va_list argptr)
+{
+    vsnprintf(msg_.data(), msg_.size(), fmt, argptr);
 }
 
 CvodeException::CvodeException(const int error_code, const char *function) :
@@ -46,6 +57,14 @@ IntegrationFailureB::IntegrationFailureB(int code, realtype t) :
 NewtonFailure::NewtonFailure(int code, const char *function) :
     AmiException("NewtonSolver routine %s failed with error code %i",function,code) {
     error_code = code;
+}
+
+SetupFailure::SetupFailure(const char *fmt, ...)
+{
+    va_list ap;
+    va_start(ap, fmt);
+    storeMessage(fmt, ap);
+    va_end(ap);
 }
 
 } // namespace amici

